Commented and removed unused code for scripts and source files

NxFerrara opened the pull request:
All scripts and source files have been commented. 

Programs modified by removing unused code: read_ultralight.c, button_listener.c, and screen.py 

Please review the commented code to become more familiar. Identify areas of needed clarification, so I can make it clearer. Merge the branch if the comments are sensible.

NxFerrara committed:
Commented and removed unused code for scripts and source files

NxFerrara committed:
Commented and removed unused code for scripts and source files

jbesoto commented on scripts/monitor.sh:
```diff
@@ -1,96 +1,99 @@
 
-# Specify the name and path of your C programs
+# Define the names and paths of the C programs to be monitored.
 PROGRAM_NAME_1="read_ultralight"
```
Consider using arrays to manage the program names and paths. This approach will make the script more concise, reduce potential errors from copy-pasting, and make it easier to add or remove programs in the future. 

```diff
- PROGRAM_NAME_1="read_ultralight"
- PROGRAM_NAME_2="button_listener"
- PROGRAM_NAME_3="screen.py"
- PROGRAM_PATH_1="$BASE_DIR/$PROGRAM_NAME_1"
- PROGRAM_PATH_2="$BASE_DIR/$PROGRAM_NAME_2"
- PROGRAM_PATH_3="$BASE_DIR/$PROGRAM_NAME_3"
```
Suggested change:
```diff
+ PROGRAM_NAMES=("read_ultralight" "button_listener" "screen.py")
+ for idx in ${!PROGRAM_NAMES[@]}; do
+    PROGRAM_PATHS[$idx]=("${BASE_DIR}/${PROGRAM_NAMES[$idx]}")
```

kc34251 replied:
Looks good

jbesoto commented on src/button_listener.c:
```diff
@@ -3,41 +3,57 @@
 
-// This function will be called when SIGINT is sent to the program (Ctrl+C)
 void handle_sigint(int sig) {
+    /**
```
Use Doxygen-style to comment functions. Here is an example from GPT:
```c
/**
 * @brief Computes the area of a rectangle.
 *
 * Given the width and height of a rectangle, this function calculates and
 * returns its area.
 *
 * @param width The width of the rectangle. Must be non-negative.
 * @param height The height of the rectangle. Must be non-negative.
 * 
 * @return The area of the rectangle. Returns -1 if any parameter is negative.
 *
 * @note This function does not check for arithmetic overflow.
 */
double computeRectangleArea(double width, double height) {
    if (width < 0 || height < 0) {
        return -1.0;  // Error value
    }
    return width * height;
}
```
The `@brief` summarizes the function while `@note` are additional notes or warnings about the function (optional).

jbesoto commented on src/read_ultralight.c:
```diff
@@ -15,18 +12,17 @@
-// The function that logs messages with timestamp
 void print_log(const char *format, ...) {
+    /**
+     * Function: print_log
```
Use Doxygen-style to document the functions.

jbesoto commented on src/screen.py:
```diff
@@ -21,13 +21,30 @@
 
 
 def handle_sigterm(signum, frame):
+    """
```
Use Google Style for commenting functions. More information [here](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods).

jbesoto commented on src/dynamodb.py:
```diff
@@ -6,17 +6,29 @@
 
-
 def delete_all_items(table):
+    """
```
Use Google-style to comment functions. More information [here](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods)

jbesoto reviewed (changes requested):
Good, meaningful comments. Fix minor details.

NxFerrara committed:
Changed src file to script file + moved file to script folder

NxFerrara committed:
Modified file structure

Modifications made to the file structure.

**Restructured Folders:**

`tests/`: Test files for source files.
`tests/ec2`: Test files for EC2 API processor source files.
`tests/iot`: Test files for IoT data collector source files.
`tests/utility`: Test files for utility source files.

`scripts/build`: Scripts for building.
`scripts/build/ec2`: Scripts specific to building the EC2 API processor.
`scripts/build/iot`: Scripts specific to building the IoT data collector.
`scripts/services`: Service scripts.
`scripts/services/ec2`: Service scripts for EC2 API processor.
`scripts/services/iot`: Service scripts for IoT data collector.
`scripts/utility`: Utility scripts.

`src/ec2`: Source files for EC2 API processor.
`src/iot`: Source files for IoT data collector.
`src/utility`: Utility source files.

**Renamed Files:**

`monitor.sh`->`watchdog.sh`: Monitors and resets core program files for IoT data collector
`configure.sh`->`build.sh`: Build file for IoT data collector
`button_listener.c`->`operation_tap_listener.c`: Listener for operation taps from touch sensor hardware module.
`read_ultralight.c`->`nfc_tap_listener.c`: Listener for NFC taps from NFC reader hardware module.
`screen.py`->`display_and_data_handler.py`: Displays data on the screen hardware module and pushes data to AWS dynamodb.
`processAPI.py`->`process_api_requests.py`: Processes API requests from AWS dynamodb to Airtable.

kc34251 commented:
Looks good

NxFerrara commented:
The most recent commit to this branch may cause some merge conflicts with the refactor/repo-cleanup branch since its most recent commit refactored by reorganizing some src and scripts files while renaming them as well. Just wanted to mention this for when merge conflict occurs.

jbesoto committed:
Fix unnecessary variable creation

jbesoto committed:
Fix unnecessary variable creation

NxFerrara commented on scripts/monitor.sh:
```diff
@@ -51,37 +52,37 @@ while true; do
 
     # Check if the second program is running.
-    if pgrep -f $PROGRAM_NAME_2 > /dev/null
+    if pgrep -f $PROGRAM_NAMES[2] > /dev/null
```
Owner, so I can't make a suggestion directly, but here is my suggestion:
`if pgrep -f $PROGRAM_NAMES[1] > /dev/null`

jbesoto replied:
Thanks for pointing it out!

NxFerrara commented on scripts/monitor.sh:
```diff
@@ -4,12 +4,13 @@
+PROGRAM_NAMES=(
+    "nfc_tap_listener.c"
+    "operation_tap_listener.c"
+    "screen.py"
```
Owner, so I can't make a suggestion directly, but here is my suggestion:
`"display_and_data_handler.py"`

jbesoto committed:
Merge branch 'main' to refactor/comment-and-clean-code-FPTS-8

jbesoto committed:
Fix typo and update file name in code

jbesoto committed:
Add newline

jbesoto commented:
Pending update comment format in Python files. I will take care of it.

jbesoto committed:
Update comment format and improve code style

jbesoto reviewed (approved):
Approving comment format. Changes previously discussed and reviewed in-person with @NxFerrara.

jbesoto committed:
Fix formatting following Google standards

jbesoto committed:
Remove private keys, replace with envars

jbesoto reviewed (approved):
- [x] Environment variables used
- [x] Commenting following team guidelines
- [x] Remove unnecessary/unused code (create new issue to fix)

jbesoto reviewed (approved):
Ignore this. Changes already approved

NxFerrara committed:
Removed Duplicate Files
